Core/BookmarksCoreDataStorage.swift

```diff
@@ -1004,4 +1013,99 @@ public class BookmarksCoreDataStorageMigration {
     }
 }
 
+
+// MARK: - CoreData structure fixer
+// https://app.asana.com/0/414709148257752/1202779945035904/f
+
+extension BookmarksCoreDataStorage {
+    
+    private func deleteExtraOrphanedFolders(_ orphanedFolders: [BookmarkFolderManagedObject],
```

Comment:
I'm still on the fence about including this at all since it doesn't seem to ever actually occur at the moment. I'm okay with it on the understanding that it's temporary, but, if we're pretty sure the issue is the other thing, including this might just muddy the waters on what actually fixed it

Reply:
I assume also this will only ever be called with one type of folder?

Reply:
Oh also, have you updated the asana task with info about which scenario actually occurs?

Reply:
@THISISDINOSAUR as for keeping both fixes and muddying the water, I don't think it's a problem since we have pixels for all the fixes as well, so we'll know if just one or both are being called.